keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -510,6 +510,9 @@ public DoubleTensor expInPlace() {
     public DoubleTensor minusInPlace(DoubleTensor that) {
         if (that.isScalar()) {
             tensor.subi(that.scalar());
+        } else if (this.isScalar()) {
```

Comment:
Is this just a problem with Scalars?  Or a wider problem with any tensor operation between Tensors of order N, M where N < M?

Reply:
Scratch that - I chatted to Caleb quickly and apparently ND4J will handle this for us under the covers.  Would be great to add a test case to ensure that works just fine though!